ml-agents/mlagents/trainers/torch/model_serialization.py

```diff
@@ -0,0 +1,70 @@
+import os
+import torch
+
+from mlagents_envs.logging_util import get_logger
+from mlagents.trainers.settings import SerializationSettings
+
+
+logger = get_logger(__name__)
+
+
+class ModelSerializer:
+    def __init__(self, policy):
+        self.policy = policy
+        dummy_vec_obs = [torch.zeros([1] + [self.policy.vec_obs_size])]
+        dummy_vis_obs = (
+            [torch.zeros([1] + list(self.policy.vis_obs_shape))]
+            if self.policy.vis_obs_size > 0
+            else []
+        )
+        dummy_masks = torch.ones([1] + self.policy.actor_critic.act_size)
+        dummy_memories = torch.zeros([1] + [self.policy.m_size])
+
+        self.input_names = [
+            "vector_observation",
+            "visual_observation",
+            "action_mask",
+            "memories",
+        ]
+        self.output_names = [
+            "action",
+            "action_probs",
+            "version_number",
+            "memory_size",
+            "is_continuous_control",
+            "action_output_shape",
+        ]
+        self.dynamic_axes = {
+            "vector_observation": [0],
+            "visual_observation": [0],
+            "action_mask": [0],
+            "memories": [0],
+            "action": [0],
+            "action_probs": [0],
+        }
+        self.dummy_input = (dummy_vec_obs, dummy_vis_obs, dummy_masks, dummy_memories)
+
+    def export_policy_model(self, output_filepath: str) -> None:
```

Comment:
I think this should be a static method, `export_policy_model(policy, output_filepath)`.

Reply:
besides policy it also needs all the dummy inputs in the class. I put them in the class so that it doesn't need to create them every time doing export.